java/feathub-connectors/flink-connectors/flink-connector-prometheus/src/main/java/com/alibaba/feathub/flink/connectors/PrometheusSinkFunction.java

```diff
@@ -0,0 +1,181 @@
+    private PushGateway pushGateway;
+    private String jobName;
+    private Map<String, String> extraLabels;
+    private Gauge[] metrics;
+    private final List<DataType> fieldTypes;
+
+    public PrometheusSinkFunction(ReadableConfig config, ResolvedSchema schema) {
+        this.config = config;
+        this.fieldTypes = schema.getColumnDataTypes();
+        this.columnNames =
+                schema.getColumns().stream().map(Column::getName).collect(Collectors.toList());
+        this.columnComments =
+                schema.getColumns().stream()
+                        .map(c -> c.getComment().orElse(c.getName()))
+                        .collect(Collectors.toList());
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+
+        registry = new CollectorRegistry();
+        pushGateway = new PushGateway(config.get(HOST_URL));
+        jobName = config.get(JOB_NAME);
+        extraLabels = parseGroupingKey(config.get(EXTRA_LABELS));
+
+        metrics = new Gauge[columnNames.size()];
+        for (int i = 0; i < columnNames.size(); i++) {
+            final LogicalType logicalType = fieldTypes.get(i).getLogicalType();
+            if (!logicalType.is(LogicalTypeFamily.NUMERIC)) {
+                throw new IOException("PrometheusSink only support numeric data types.");
+            }
+
+            metrics[i] =
+                    Gauge.build()
+                            .name(columnNames.get(i))
+                            .help(columnComments.get(i))
+                            .register(registry);
+        }
+    }
+
+    @Override
+    public void invoke(RowData row, Context context) throws Exception {
+        for (int i = 0; i < metrics.length; i++) {
+            final Gauge metric = metrics[i];
+            final LogicalType type = fieldTypes.get(i).getLogicalType();
+            switch (type.getTypeRoot()) {
+                case DECIMAL:
+                    final int precision = ((DecimalType) type).getPrecision();
+                    final int scale = ((DecimalType) type).getScale();
+                    final DecimalData decimal = row.getDecimal(i, precision, scale);
+                    metric.set(decimal.toBigDecimal().doubleValue());
+                    break;
+                case TINYINT:
+                    metric.set(row.getByte(i));
+                    break;
+                case SMALLINT:
+                    metric.set(row.getShort(i));
+                    break;
+                case INTEGER:
+                    metric.set(row.getInt(i));
+                    break;
+                case BIGINT:
+                    metric.set(row.getLong(i));
+                    break;
+                case FLOAT:
+                    metric.set(row.getFloat(i));
+                    break;
+                case DOUBLE:
+                    metric.set(row.getDouble(i));
+                    break;
+                default:
```

Comment:
In some scenarios, a feature might only have a limited number of possible values, and users want to count the ratio for each label. 

In order to support this scenario, let's also add support for `Map<String, Number>` types. Each entry corresponds to a metric in Prometheus, the key in an entry is a label of the metric, and the value of the entry is the value of the metric.